app/controllers/attributes_controller.rb

```diff
@@ -0,0 +1,44 @@
+class AttributesController < ApplicationController
```

Comment:
Just checking we're doing a lot of pulling keys out of hashes in this controller.

They all seem to be coming from params or http-recieved objects that rails has passed.

The keys here are all symbols. I remember this catching us out in the past, no?

Are we sure that JSON processed string key names get processed into symbols?

Reply:
> Are we sure that JSON processed string key names get processed into symbols?

Actually they're all strings, no symbols involved at all.